### PR TITLE
feat(application): add CreateBookUseCase with orchestration logic

### LINKED ISSUE #6: Implementar adaptador OllamaEmbeddingService
## Descripción

Implementar el adaptador que conecta con Ollama para generar embeddings.

## Tareas

- [ ] Crear `src/infrastructure/driven/embedding/OllamaEmbeddingService.ts`
- [ ] Implementar interfaz `EmbeddingService`
- [ ] Configurar llamada HTTP a Ollama API (`POST /api/embeddings`)
- [ ] Manejar timeout (30 segundos recomendado)
- [ ] Manejar errores de conexión → `EmbeddingServiceUnavailableError`
- [ ] Validar longitud del texto (≤ 7000 chars) → `EmbeddingTextTooLongError`
- [ ] Añadir configuración en `src/infrastructure/config/env.ts`
- [ ] Tests unitarios con mocks de fetch

## Configuración Ollama

| Parámetro | Valor |
|-----------|-------|
| Modelo | `nomic-embed-text` |
| Endpoint | `POST http://ollama:11434/api/embeddings` |
| Dimensión | 768 |
| Timeout | 30s |

## Request/Response

```typescript
// Request
{
  "model": "nomic-embed-text",
  "prompt": "texto a vectorizar..."
}

// Response
{
  "embedding": [0.123, -0.456, ...]  // 768 dimensiones
}
```

## Criterios de Aceptación

- [ ] Genera embeddings correctamente llamando a Ollama
- [ ] Maneja errores de conexión con mensaje claro
- [ ] Valida longitud máxima de texto
- [ ] Tests unitarios con cobertura ≥ 80%
- [ ] Configuración externalizada (URL, timeout, modelo)

## Dependencias

- Requiere: #3 (Puerto EmbeddingService)

## Parent Issue

Parte de #1 (HU-001: Registro de Libros)

### PR DESCRIPTION
**Pull request by albixu**
## Summary
- Implements `CreateBookUseCase` that orchestrates the complete book creation flow: category resolution, validation, duplicate checking, embedding generation, and atomic persistence
- Adds 11 comprehensive unit tests covering success paths, error handling, and edge cases

## Changes
- `apps/api-cli/src/application/use-cases/CreateBookUseCase.ts` - Main use case implementation
- `apps/api-cli/src/application/use-cases/index.ts` - Barrel export
- `apps/api-cli/src/application/index.ts` - Export use-cases module
- `apps/api-cli/tests/unit/application/use-cases/CreateBookUseCase.test.ts` - Unit tests

## Closes
Closes #6

### CODE REVIEW

File: apps/api-cli/src/application/use-cases/CreateBookUseCase.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+   * @throws EmbeddingTextTooLongError if embedding text exceeds 7000 chars
+   * @throws EmbeddingServiceUnavailableError if embedding service is down
+   * @throws DomainError for validation failures
+   */
+  async execute(input: CreateBookInput): Promise<CreateBookOutput> {
+    // 1. Resolve or create categories
+    const categories = await this.categoryRepository.findOrCreateMany(
+      input.categoryNames
+    );
+
```
[the remark this answers is not in the document]
Comment (albixu) [reply apps/api-cli/src/application/use-cases/CreateBookUseCase.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/albixu/library/pull/32#discussion_r2779341322)

File: apps/api-cli/src/application/use-cases/CreateBookUseCase.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    // 4. Generate embedding text and validate length
+    const embeddingText = book.getTextForEmbedding();
+
+    if (embeddingText.length > MAX_EMBEDDING_TEXT_LENGTH) {
+      throw new EmbeddingTextTooLongError(
+        embeddingText.length,
+        MAX_EMBEDDING_TEXT_LENGTH
+      );
+    }
```
[the remark this answers is not in the document]
Comment (albixu) [reply apps/api-cli/src/application/use-cases/CreateBookUseCase.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/albixu/library/pull/32#discussion_r2779341331). Document why it's kept and add a realistic way to test it

File: apps/api-cli/tests/unit/application/use-cases/CreateBookUseCase.test.ts [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+    it('should throw BookAlreadyExistsError when triad duplicate found', async () => {
+      const duplicateResult: DuplicateCheckResult = {
+        isDuplicate: true,
+        duplicateType: 'triad',
+        message: 'A book with the same author, title, and format already exists',
+      };
+      (mockBookRepository.checkDuplicate as ReturnType<typeof vi.fn>).mockResolvedValue(duplicateResult);
+
+      await expect(useCase.execute(validInput)).rejects.toThrow(BookAlreadyExistsError);
+    });
```
[the remark this answers is not in the document]
Comment (albixu) [reply apps/api-cli/tests/unit/application/use-cases/CreateBookUseCase.test.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/albixu/library/pull/32#discussion_r2779341365)

File: apps/api-cli/src/application/use-cases/CreateBookUseCase.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
+/**
+ * CreateBookUseCase
+ *
+ * Application service that orchestrates the book creation process.
+ * This use case coordinates between domain entities and infrastructure ports.
```
[the remark this answers is not in the document]
Comment (albixu) [reply apps/api-cli/src/application/use-cases/CreateBookUseCase.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/albixu/library/pull/32#discussion_r2779341372)

File: apps/api-cli/src/application/use-cases/CreateBookUseCase.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+      author: book.author,
+      title: book.title,
+      format: book.format.value,
+    });
+
+    if (duplicateCheck.isDuplicate) {
+      throw new BookAlreadyExistsError(
+        duplicateCheck.message ?? 'Duplicate book found'
+      );
+    }
```
[the remark this answers is not in the document]
Comment (albixu) [reply apps/api-cli/src/application/use-cases/CreateBookUseCase.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/albixu/library/pull/32#discussion_r2779341375)
